### ISSUE TITLE
from_pulseq drops TM: a stimulated-echo sequence loses its mixing time on export→import

### ISSUE THREAD

**Issue by rutgerfick** [dmrai-lab/dmipy-sim#28]
`Waveform` carries `TM` (and `stimulated_echo`), and `set_b`/`rotate_waveform` correctly propagate them. But the Pulseq bridge does not round-trip `TM`:

```python
# sequences/pulseq.py::from_pulseq
return Waveform(G=jnp.asarray(G[None]), dt=dt, echo_idx=echo_idx, rf_events=rf_events)
#                                                                  ^ no TM=
```

So `to_pulseq(w)` followed by `from_pulseq(...)` returns a waveform whose `TM` is `None`. For a PGSTE/STE sequence the mixing time is the entire point of the acquisition — it is what makes the measurement T1-weighted and what sets the diffusion time — and losing it is silent: the gradient waveform still round-trips, the b-value still round-trips, and nothing raises.

Anything downstream that branches on `TM`/`stimulated_echo` (the T1 path in the step function, pathway sign handling, replay envelope summaries) then silently takes the spin-echo branch for a stimulated-echo acquisition.

### Fix

Carry it through a custom Pulseq definition (e.g. `dmipy_TM`): write it in `to_pulseq`, read it back in `from_pulseq`, leaving `None` when absent so non-dmipy `.seq` files are unaffected.

### Test

A round-trip assertion over a PGSTE waveform: `TM`, `stimulated_echo`, `echo_idx` and `dt` all survive `to_pulseq → from_pulseq`. Worth parameterising over the sequence families the bridge claims to support, since a missing-field bug of this shape is invisible unless asserted field by field.
